docs-search/common/src/Docs/Search/TypeIndex.purs

```diff
@@ -80,15 +84,22 @@ lookup
   -> Aff { index :: TypeIndex, results :: Array SearchResult }
 lookup key index@(TypeIndex map) =
   case Map.lookup key map of
-    Just results -> pure { index, results: fold results }
+    Just results ->
+      pure { index, results: fold results }
     Nothing -> do
-      eiJson <- try (toAffE (lookup_ key $ Config.mkShapeScriptPath key))
-      pure $ fromMaybe'
-        (\_ -> { index: insert key Nothing index, results: [] })
-        do
-          json <- hush eiJson
-          results <- hush (CJ.decode (CJ.array SearchResult.searchResultCodec) json)
+      eitherJson <- try $ toAffE $ lookup_ key (Config.mkShapeScriptPath key)
+
+      let
+        eitherResults = do
+          json <- eitherJson # lmap Error.message
+          CJ.decode (CJ.array SearchResult.searchResultCodec) json # lmap DecodeError.print
+
+      case eitherResults of
+        Right results ->
           pure { index: insert key (Just results) index, results }
+        Left err -> do
+          Console.error $ "Error reading type index: " <> err
+          pure { index: insert key Nothing index, results: [] }
```

Comment:
This is where the UI is trying to read an array value from the index JSON. I did not change how it functions in principle, but added an error message to the browser console if load or parsing fails. So that at least we can see what's happening.